### DISCUSSION TITLE
Mesh - getParticles()

### DISCUSSION THREAD

**Discussion by HarshithaBR** [empriselab/RCareWorld#121] [Help - Programming] [answered]
Dear Organizers,

I am trying to get a mesh for the cloth in the dressing task, and have been following `example_cloth_attach.py`.

I have modified `test_dressing.py` and I am trying to check the data and get the particles from the mesh.
Q1) What is the best way to get the path for the cloth mesh.
```
     t_shirt_path = os.path.join(mesh_path, 'Tshirt.obj')
     mesh = env.LoadCloth(path=t_shirt_path)
```

Q2) If I print `mesh` and `mesh.data`, I am getting the below. Can you please suggest on how to work with the same?
```
type(mesh):  <class 'pyrcareworld.attributes.cloth_attr.ClothAttr'>`

mesh.data: {}
```

hence, `position1 = mesh.data['particles'][500]` would throw an error.

Thanks and best Regards
Harshitha B R

**Comment by YoruCathy** [comment]
Hi @HarshithaBR does this example work for you? https://github.com/empriselab/RCareWorld/blob/phy-robo-care/pyrcareworld/pyrcareworld/demo/examples/example_cloth_attach.py

**Reply by HarshithaBR** [reply]
Hi Ruolin

It was working for me till yesterday. I was able to visualize the picking of t-shirt and swaying of the same.

However, I am getting an error now.

Please find attached the logs for the run.
[player_logs.log](https://github.com/user-attachments/files/17020025/player_logs.log)

Thanks and regards
Harshitha

**Reply by YoruCathy** [reply]
Did you do any git pull to update the codebase?

**Reply by HarshithaBR** [reply]
yes, I did a git pull on Sunday.

**Reply by YoruCathy** [reply]
Which commit is it at before you did the pull? I highly doubt some local changes might have affected this since we haven't updated the example player for a while.

**Reply by HarshithaBR** [reply]
Hi

I was at commit : 30d03bf5469107fa798c14b16e7d89ecedbdeac4
before i did the git pull

**Reply by YoruCathy** [reply]
Except for doing the pull, did you install anything else? The newest commit works on my side. You can try rebooting your system and see if it works. If it still doesn't work, you can try installing the environment from scratch. Especially, see if these commands will help: 
```
sudo apt-get update
sudo apt-get install libassimp-dev libopenblas-dev liblapack-dev
```
It looks like the libraries are not working.

**Reply by HarshithaBR** [reply]
okay, i will try it out

Thank you!

**Reply by frankthetank91** [reply]
Hello!

I also ran into the same issue. The 'example_cloth_attach.py' works for me but once I run 'cloth.getParticles()', it returns None in the 'test_dressing.py --graphics'. Would you be able to check this? I installed today with the latest update on Ubuntu 20.04

**Reply by YoruCathy** [reply]
Hi, we have the unit tests for this function working. Could you both try if it works on your side? https://github.com/empriselab/RCareWorld/blob/phy-robo-care/tests/test_dressing_env.py

**Reply by YoruCathy** [reply, marked as answer]
Works and tested.
